Frameworks/Foundation/NSDecimalNumber.mm

```diff
@@ -14,296 +14,344 @@
-                                     NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)maximumDecimalNumber {
+    static NSDecimalNumber* s_maxNumber =
+        [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::max() exponent:_MAX_EXP isNegative:NO];
+    return s_maxNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)minimumDecimalNumber {
+    static NSDecimalNumber* s_minNumber =
+        [[NSDecimalNumber alloc] initWithMantissa:std::numeric_limits<unsigned long long>::max() exponent:_MAX_EXP isNegative:YES];
+    return s_minNumber;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalNormalize(NSDecimal* number1, NSDecimal* number2, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (NSDecimalNumber*)decimalNumberWithDecimal:(NSDecimal)decimal {
+    return [[[NSDecimalNumber alloc] initWithDecimal:decimal] autorelease];
+}
+
+/**
+ @Status Interoperable
+*/
++ (NSDecimalNumber*)decimalNumberWithMantissa:(unsigned long long)mantissa exponent:(short)exponent isNegative:(BOOL)isNegative {
+    return [[[NSDecimalNumber alloc] initWithMantissa:mantissa exponent:exponent isNegative:isNegative] autorelease];
 }
 
 /**
  @Status Stub
  @Notes
 */
-NSCalculationError NSDecimalPower(NSDecimal* result, const NSDecimal* number, NSUInteger power, NSRoundingMode roundingMode) {
++ (NSDecimalNumber*)decimalNumberWithString:(NSString*)numericString locale:(id)locale {
     UNIMPLEMENTED();
     return StubReturn();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-void NSDecimalRound(NSDecimal* result, const NSDecimal* number, NSInteger scale, NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
+- (NSNumber*)initWithBool:(BOOL)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSString* NSDecimalString(const NSDecimal* dcm, id locale) {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithChar:(char)value {
+    return [self initWithLongLong:static_cast<long long>(value)];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-NSCalculationError NSDecimalSubtract(NSDecimal* result,
-                                     const NSDecimal* leftOperand,
-                                     const NSDecimal* rightOperand,
-                                     NSRoundingMode roundingMode) {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSNumber*)initWithDouble:(double)value {
+    int exp = 0;
+    double res = value;
+    if (res < 0) {
+        res *= -1;
+    }
+
+    if (res == DBL_MIN) {
+        return [self initWithMantissa:0 exponent:0 isNegative:0];
```

Comment:
wait what? I'm assuming that this is 1 * 10^-(big number). Is (big number) > 128?